src/components/lib/cos_kernel_api.c

```diff
@@ -421,7 +422,7 @@ cos_captbl_alloc(struct cos_compinfo *ci)
 	assert(ci);
 
 	if (__alloc_mem_cap(ci, CAP_CAPTBL, &kmem, &cap)) return 0;
-	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, ci->pgtbl_cap, kmem, 0)) BUG();
+	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, __compinfo_metacap(ci)->local_mi.pgtbl_cap, kmem, 0)) BUG();
```

Comment:
@gparmer it worked fine before because the threads were created from `vkernel` component and were copied to vm captbl, which as you said didn't use `meta`. And when the VMs started, they had their own untyped memory, so again the same no `meta` case there. 

@ryuxin this looks alright to me, but i've come to realize all the `ci->pgtbl_cap` are going to be replaced with `ci->local_mi.pgtbl_cap`, which pretty much removes `SELF_PT` from the picture. Isn't it?